examples/interpreter/physl.cpp

```diff
@@ -45,6 +47,14 @@ std::string read_user_code(std::string const& path)
     return str_stream.str();
 }
 
+std::tuple<bool, std::string> get_env_physl_ir() {
+    auto e = boost::this_process::environment();
+    auto env_physl_ir = e.find("PHYSL_IR");
+    if(env_physl_ir == e.end()) { return std::make_tuple(false, std::string{}); }
+    auto physl_ir_str = boost::beast::detail::base64_decode(env_physl_ir.second);
+    return std::make_tuple(true, physl_ir_str);
```

Comment:
May I suggest to actually have the Base64 decoding included in this file directly? We try to reduce dependencies on Boost and this would add another Boost library to the mix. Aditionally, Boost.Beast was added to Boost just recently, so this wouldn't compile for older versions.

Here is the code:
```
std::string base64_decode(std::string const& encoded_string) {
  int in_len = encoded_string.size();
  int i = 0;
  int j = 0;
  int in_ = 0;
  unsigned char char_array_4[4], char_array_3[3];
  std::string ret;

  while (in_len-- && ( encoded_string[in_] != '=') && is_base64(encoded_string[in_])) {
    char_array_4[i++] = encoded_string[in_]; in_++;
    if (i ==4) {
      for (i = 0; i <4; i++)
        char_array_4[i] = base64_chars.find(char_array_4[i]);

      char_array_3[0] = ( char_array_4[0] << 2       ) + ((char_array_4[1] & 0x30) >> 4);
      char_array_3[1] = ((char_array_4[1] & 0xf) << 4) + ((char_array_4[2] & 0x3c) >> 2);
      char_array_3[2] = ((char_array_4[2] & 0x3) << 6) +   char_array_4[3];

      for (i = 0; (i < 3); i++)
        ret += char_array_3[i];
      i = 0;
    }
  }

  if (i) {
    for (j = 0; j < i; j++)
      char_array_4[j] = base64_chars.find(char_array_4[j]);

    char_array_3[0] = (char_array_4[0] << 2) + ((char_array_4[1] & 0x30) >> 4);
    char_array_3[1] = ((char_array_4[1] & 0xf) << 4) + ((char_array_4[2] & 0x3c) >> 2);

    for (j = 0; (j < i - 1); j++) ret += char_array_3[j];
  }

  return ret;
}
```

Reply:
Will add that into the commit. Should that code have a home in physl.cpp or another file?

Reply:
For now I think it's fine to have it right here. We'll go and move it if we come across another use somewhere else.